File: WIS/source/WIS_Prototype/WIS_PrototypeAPI/Data/Models/Load.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WIS_PrototypeAPI.Data.Models
{
	public class Load
	{
		[Key]
		[Required]
		public long LoadId { get; set; }

		public int? GrossWeight {  get; set; }
		public int? TareWeight { get; set; }
		public int? NetWeight { get; set; }
		[Column(TypeName = "nvarchar(20)")]
		public string? TruckId { get; set; } = null;
		public DateTime? TimeIn { get; set; } = null;
		public DateTime? TimeOut { get; set;} = null;
		[Column(TypeName = "nvarchar(20)")]
		public string? BillOfLading { get; set; } = null;
		public double? MoistureLevel { get; set; }
		public double? TestWeight { get; set; }
		public double? ProtienLevel { get; set; }
		[Column(TypeName = "nvarchar(200)")]
		public string? Notes { get; set; } = null;

		// many to one
		[ForeignKey(nameof(Weightsheet))]
		public long? WeightsheetIdLink { get; set; }
		public Weightsheet? Weightsheet { get; set; } = null;
	}
}

File: WIS/source/WIS_Prototype/WIS_Prototype/src/app/load/load.ts
// Filename: load.ts
// Purpose: Allow for modularity and ability to send HTTP POST request.

// ILoad parent class
// Purpose: Define a common interface for other Load classes to inhert from. 
export class ILoad {
  loadId?: bigint;
  weightsheetIdLink?: bigint;
  grossWeight: number;
  tareWeight: number;
  netWeight: number;
  truckId: string;
  timeIn?: Date;
  bolNumber?: number;
  destBin?: number;
  notes?: string;

  constructor() {
    this.grossWeight = 0
    this.tareWeight = 0;
    this.netWeight = 0;
    this.truckId = '';
  }
}

// Load Class
// Purpose: Defines state for GET requets for Loads. Hass all load properties
export class Load extends ILoad{
  timeOut?: Date;
  moistureLevel?: number;
  testWeight?: number;
  protienLevel?: number;
}


export class LoadMoistureTestWeightProtien extends ILoad {
  moistureLevel?: number;
  testWeight?: number;
  protienLevel?: number;
}

export class LoadMoisture extends ILoad {
  moistureLevel?: number;
}

export class LoadTestWeight extends ILoad {
  testWeight?: number;
}

export class LoadProtien extends ILoad {
  protienLevel?: number;
}

export class LoadMoistureTestWeight extends ILoad {
  moistureLevel?: number;
  testWeight?: number;
}

export class LoadMoistureProtien extends ILoad {
  moistureLevel?: number;
  protienLevel?: number;
}

export class LoadTestWeightProtein extends ILoad {
  testWeight?: number;
  protienLevel?: number;
}
